core/src/commons/Data.h

```diff
@@ -55,7 +55,7 @@ class Data {
 
   void set_weight_index(size_t index);
 
-  void get_all_values(std::vector<double>& all_values, const std::vector<size_t>& samples, size_t var) const;
+  std::vector<size_t> get_all_values(std::vector<double>& all_values, const std::vector<size_t>& samples, size_t var) const;
```

Comment:
I think it would be clearer + more consistent to return the sorted samples through a parameter, since we already have one output parameter `std::vector<double>& all_values`. Also it'd be great to add a method comment, it's not really clear what the two return values represent from looking at the signature.